src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -111,9 +123,18 @@ private static float CosineSimilarityCore(ReadOnlySpan<float> x, ReadOnlySpan<fl
                     Vector256<float> xVec = Vector256.LoadUnsafe(ref xRef, (uint)i);
                     Vector256<float> yVec = Vector256.LoadUnsafe(ref yRef, (uint)i);
 
-                    dotProductVector += xVec * yVec;
-                    xSumOfSquaresVector += xVec * xVec;
-                    ySumOfSquaresVector += yVec * yVec;
+                    if (Fma.IsSupported)
```

Comment:
Is this check needed inside of Vector256.IsHardwareAccelerated?

Reply:
--instruction-set avx2 doesn't enable FMA but enables Vector256.IsHardwareAccelerated so yes, this check is needed as avx2 doesn't imply FMA